### ISSUE TITLE
XSPICEPS1S2 transformer issues

### ISSUE THREAD

**Issue by tomhajjar** [ra3xdh/qucs_s#367]
I found more problems with XSPICEPS1S2 in Transformer.lib. I made three circuits comparing XSPICEPS1S2 and TransformerPS1S2 in Transformer.lib to Three Mutual Inductors in Components. Analysis of TransformerPS1S2 and Three Mutual Inductors generally agree but XSPICEPS1S2 is wrong and generates errors.

XSPICEPS1S2 has two issues:

1. Rp, Rs1 and Rs2 generate warnings. Is there node naming issues like in TransformerPS1S2?
2. The "Turns Ratio" is half what it should be. You can see the output voltages and currents in the plots are half compared to transformers TransformerPS1S2 and Three Mutual Inductors.

I recreated the subcircuit for XSPICEPS1S2 made by Mike Brinson. It can be modified to fix the "turns ratio" problem.

I also recreated the subcircuit for XSPICEPS1. It's "turns ratio" seems OK.

**TODO**

- [x] Replace models in Transformer.lib

![2023-11-15_192246](https://github.com/ra3xdh/qucs_s/assets/102838604/938d9df8-6721-4121-bbe2-65dcaf4710c2)

![2023-11-15_205939](https://github.com/ra3xdh/qucs_s/assets/102838604/0efb190d-a981-4d00-9d15-560394a10d16)
 
![2023-11-15_215510](https://github.com/ra3xdh/qucs_s/assets/102838604/ff8e9a32-5805-46ba-8418-3c45cb79572a)

[Transformer_Lib_Test_prj.zip](https://github.com/ra3xdh/qucs_s/files/13371833/Transformer_Lib_Test_prj.zip)

**Comment by ra3xdh** [comment]
> Rp, Rs1 and Rs2 generate warnings. Is there node naming issues like in TransformerPS1S2?

No, the variable should be wrapped in braces like `{Rs1}` Otherwise Ngspice may interpret it as the model.

**Comment by tomhajjar** [comment]
Holger corrected the XSPICEPS1S2 subcircuit. AFAIK, the info in the online help is wrong.

I found another issue, Cores.lib uses the same data for every core so only one core is correct....

**Comment by ra3xdh** [comment]
> Cores.lib uses the same data for every core so only one core is correct..

Yes, I have also noticed that the Cores.lib uses duplicate data. Unfortunately, I have no source data for this library and cannot provide the fix. The library was developed by @MikeBrinson around 2015. I cannot say where did he take the cores data.

**Comment by ra3xdh** [comment]
> AFAIK, the info in the online help is wrong.

The help on readthedocs.io should be deprecated and replaced by the new tutorial. It contains a lot of outdated information. I have started to collect the new documentation here: https://github.com/ra3xdh/qucs_s_docs But this a slow process. Writing documentation requires much more time effort than writing a code.

**Comment by tomhajjar** [comment]
Here is the updated Transformers.lib. All identified errors have been fixed.

I left the "Cores" data in Transformers.lib and Cores.lib as it was. I could not find any info on the web. Looks like manually pulling data from B/H plots would be required. Some of the cores are nothing a user would ever make a transformer.

I requested info from Mike Brinson...

[Transformers.zip](https://github.com/ra3xdh/qucs_s/files/13383341/Transformers.zip)
